### PR TITLE
manim-slides: init at 5.1.3

### PR DESCRIPTION
**Pull request by dmadisetti**
## Description of changes

Another implementation fixing #251914 (edit: wrong issue linked earlier)

**NOTE**: Work in progress because ignoring some `manim` tests, and rather clumsily put together.

More done to explore deficiencies in #275577

## Things done

 - Fixed Manim
 - got manim-slides working with pdm
 - Tested it out: https://static.marimo.app/static/manim-example-rket
 with `nix-shell -p ffmpeg manim "python3.withPackages(ps: with ps; [ marimo manim-slides ])"`

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by dmadisetti** [comment]
Also, only got it reliably working with python311, pdm did not seem to like python310 or python312

File: pkgs/development/python-modules/manim-slides/default.nix [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+  format = "pyproject";
+  version = "5.1.3";
+
+  disabled = pythonOlder "3.9";
+
+  src = fetchFromGitHub {
+    owner = "jeertmans";
+    repo = "manim-slides";
+    rev = "refs/tags/v${version}";
+    sha256 = "sha256-WZR95swapT2Fbu6mbuHLjMu3Okq/wKFabzN7xpZw0/g=";
```
Comment (soispha) [inline pkgs/development/python-modules/manim-slides/default.nix:47]: This should just be `hash`, as it is already obvious, that that's a sha256 by the `sha256`  in the hash string.

File: pkgs/development/python-modules/manim-slides/default.nix [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    requests
+    rich
+    rtoml
+    tqdm
+
+    # avconv is a potential alternative
+    ffmpeg
+    # This could also be manimgl, but that is not (yet) packaged
+    manim
+  ] ++ (if ! withGui then [ ipython ] else [ ]);
```
Comment (soispha) [inline pkgs/development/python-modules/manim-slides/default.nix:84]: Change this to `lib.lists.optional withGui ipython;`. This use-case is already covered in lib

File: pkgs/development/python-modules/manim-slides/default.nix [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+  pythonImportsCheck = [ "manim_slides" ];
+
+  meta = with lib; {
```
Comment (soispha) [inline pkgs/development/python-modules/manim-slides/default.nix:88]: Add `changelog` and `mainProgram`

File: pkgs/applications/video/manim/default.nix [hunk 4]
```diff
@@ -64,8 +64,10 @@ in python.pkgs.buildPythonApplication rec {
 
   pythonRelaxDeps = [
```
Comment (soispha) [inline pkgs/applications/video/manim/default.nix:65]: The version locks on these dependencies seem to [already](https://github.com/ManimCommunity/manim/commit/61a2d05b69862a49692bef01e776ff305beadf39)  have been relaxed on the upstream branch ([watchdog](https://github.com/ManimCommunity/manim/blob/main/pyproject.toml#L57) and [isosurfaces](https://github.com/ManimCommunity/manim/blob/main/pyproject.toml#L36)). We should probably have a comment here, noting to change that, when a new release is made.

**Review by soispha — CHANGES_REQUESTED** [review]
Thank you for improving my pr! I already have had a few changes locally, but was too busy in the last few weeks to push them, thus I just reviewed your pr and added the most significant changes in  [this commit](https://github.com/NixOS/nixpkgs/pull/275577/commits/273ca9485ad8364c2efd351fafc7f7761645e544) to my pr. You should be able to simple apply the [patch](https://github.com/NixOS/nixpkgs/pull/275577/commits/273ca9485ad8364c2efd351fafc7f7761645e544.patch).

**Comment by soispha** [comment]
I've just tested the resulting binary and it seems to be missing qt bindings. 
Running `nix-build -A manim-slides --impure` in this branch, then `./result/bin/manim-slides init` returns this error:
```
# <snip>
File "/nix/store/k4rqq4x6h1xg91asc4fm6r9s876jklb4-python3.11-qtpy-2.4.0/lib/python3.11/site-packages/qtpy/__init__.py", line 287, in <module>
    raise QtBindingsNotFoundError from None
qtpy.QtBindingsNotFoundError: No Qt bindings could be found
```

I'll try to fix this in my pr. But we should probably decide on one pr to close, to avoid having duplicates.

**Comment by soispha** [comment]
Okay [that](https://github.com/NixOS/nixpkgs/pull/275577/commits/a56ece4ee151c3eb620961c6e9ef0b9eaa57cd4a) commit resolves that.

**Comment by dmadisetti** [comment]
Thanks! Closed in favor of #275577

**Pull request closed without merging**